src/Mono.Android/Java.Interop/JavaConvert.cs

```diff
@@ -275,15 +275,15 @@ static class JavaConvert {
 		}
 
 		static Dictionary<Type, Func<object, IJavaObject>> JavaObjectConverters = new Dictionary<Type, Func<object, IJavaObject>>() {
-			{ typeof (bool),   value => new Java.Lang.Boolean ((bool) value) },
-			{ typeof (byte),   value => new Java.Lang.Byte ((sbyte) (byte) value) },
-			{ typeof (sbyte),  value => new Java.Lang.Byte ((sbyte) value) },
-			{ typeof (char),   value => new Java.Lang.Character ((char) value) },
-			{ typeof (short),  value => new Java.Lang.Short ((short) value) },
-			{ typeof (int),    value => new Java.Lang.Integer ((int) value) },
-			{ typeof (long),   value => new Java.Lang.Long ((long) value) },
-			{ typeof (float),  value => new Java.Lang.Float ((float) value) },
-			{ typeof (double), value => new Java.Lang.Double ((double) value) },
+			{ typeof (bool),   value => Java.Lang.Boolean.ValueOf ((bool) value) },
```

Comment:
The changes are *probably* safe, in that I can't easily tell if there are any callers that will be `Dispose()`ing of the value…